src/irr/asset/CMeshManipulator.cpp

```diff
@@ -1551,6 +1551,343 @@ core::smart_refctd_ptr<ICPUBuffer> IMeshManipulator::idxBufferFromTrianglesFanTo
+        findExternalPointProj(m1, dMin.z, dMax.z);
+
+        len = dMax - dMin;
+        quality = len.x * len.y + len.x * len.z + len.y * len.z;
+
+        if (quality < bestVal)
+        {
+            bestVal = quality;
+            b = core::matrix3x4SIMD(v1, n, m1);
+        }
+
+        findExternalPointProj(v2, dMin.x, dMax.x);
+        findExternalPointProj(m2, dMin.z, dMax.z);
+
+        len = dMax - dMin;
+        quality = len.x * len.y + len.x * len.z + len.y * len.z;
+
+        if (quality < bestVal)
+        {
+            bestVal = quality;
+            b = core::matrix3x4SIMD(v2, n, m2);
+        }
+    };
+
+    //find the best axes
+    {
+        //from base triangle
+        findImprovedAxesFromTriangle(e0, e1, e2, n);
+
+        //from top tetrahedra
+        core::vectorSIMDf f0 = core::normalize(t0 - p0);
+        core::vectorSIMDf f1 = core::normalize(t0 - p1);
+        core::vectorSIMDf f2 = core::normalize(t0 - p2);
+        core::vectorSIMDf n0 = core::normalize(core::cross(f1, e0));
+        core::vectorSIMDf n1 = core::normalize(core::cross(f2, e1));
+        core::vectorSIMDf n2 = core::normalize(core::cross(f0, e2));
+        /*findImprovedAxesFromTriangle(e0, f1, f0, n0);
+        findImprovedAxesFromTriangle(e1, f2, f1, n1);
+        findImprovedAxesFromTriangle(e2, f0, f2, n2);*/
+
+
+        //from bottom tetrahedra
+        f0 = core::normalize(t1 - p0);
+        f1 = core::normalize(t1 - p1);
+        f2 = core::normalize(t1 - p2);
+        n0 = core::normalize(core::cross(f1, e0));
+        n1 = core::normalize(core::cross(f2, e1));
+        n2 = core::normalize(core::cross(f0, e2));
+
+        /*findImprovedAxesFromTriangle(e0, f1, f0, n0);
+        findImprovedAxesFromTriangle(e1, f2, f1, n1);
+        findImprovedAxesFromTriangle(e2, f0, f2, n2);*/
+    }
+
+    core::matrix3x4SIMD resultRotation;
+    resultRotation[0] = core::vectorSIMDf(b[0].x, b[1].x, b[2].x);
+    resultRotation[1] = core::vectorSIMDf(b[0].y, b[1].y, b[2].y);
+    resultRotation[2] = core::vectorSIMDf(b[0].z, b[1].z, b[2].z);
+    b = resultRotation;
+
+    //compute OBB dimensions
+    {
+        //b is an orthonormal matrix, which represent rotation of the bounding box
+        bMin.x = bMax.x = core::dot(mb->getPosition(0u), b[0]).x;
+        bMin.y = bMax.y = core::dot(mb->getPosition(0u), b[1]).x;
+        bMin.z = bMax.z = core::dot(mb->getPosition(0u), b[2]).x;
+
+        for (size_t i = 1u; i < mb->calcVertexCount(); i++)
+        {
+            const core::vectorSIMDf vtxPos = mb->getPosition(i);
+            for (uint32_t j = 0u; j < 3u; j++)
+            {
+                float proj = core::dot(vtxPos, b[j]).x;
+
+                if (proj > bMax[j])
+                    bMax[j] = proj;
+
+                if (proj < bMin[j])
+                    bMin[j] = proj;
+            }
+        }
+
+        bLen = bMax - bMin;
+
+        bestVal = bLen.x * bLen.y + bLen.x * bLen.z + bLen.y * bLen.z;
+    }
```

Comment:
is `resultRotation` local OBB space to world space matrix or the other way round?

Reply:
local to world space